### PR TITLE
Use AppCheckInterop APIs

### PR DESCRIPTION
**Pull request by aashishpatil-g**
Use Interop APIs and remove hard dependency for AppCheck framework at SDK level

### CODE REVIEW

File: Package.swift [hunk 1]
```diff
@@ -45,9 +45,8 @@ let package = Package(
       dependencies: [
         .product(name: "GRPC", package: "grpc-swift"),
         .product(name: "FirebaseCore", package: "firebase-ios-sdk"),
-        // TODO: Investigate switching Auth and AppCheck to interop.
+        // TODO: Investigate switching Auth to interop.
         .product(name: "FirebaseAuth", package: "firebase-ios-sdk"),
-        .product(name: "FirebaseAppCheck", package: "firebase-ios-sdk"),
         .product(name: "GULEnvironment", package: "GoogleUtilities"),
```
Comment (paulb777) [inline Package.swift:50]: Add a dependency for FirebaseAppCheckInterop so it's not relying upon its transitive dependency through Auth

Comment (aashishpatil-g) [reply Package.swift:50]: Maybe we could use it transitively via Auth for now and then switch to it when we decide on the course forward.

Comment (paulb777) [reply Package.swift:50]: SGTM

Comment (paulb777) [reply Package.swift:50]: Would you add a comment/TODO?

**Review by paulb777 — APPROVED** [review]
LGTM on green and AppCheck manual test

File: Package.swift (hunk 1 above)
Comment (aashishpatil-g) [reply Package.swift:52]: Added a TODO in Package.swift.

**Comment by aashishpatil-g** [comment]
Confirmed both failure and success cases manually.

**Pull request merged**